crates/uv/src/commands/project/add.rs

```diff
@@ -264,6 +279,78 @@ pub(crate) async fn add(
         Err(err) => return Err(err.into()),
     };
 
+    // Avoid modifying the user request further if `--raw-sources` is set.
+    if !raw_sources {
+        // Extract the minimum-supported version for each dependency.
+        let mut minimum_version =
+            FxHashMap::with_capacity_and_hasher(lock.lock.distributions().len(), FxBuildHasher);
+        for dist in lock.lock.distributions() {
+            let name = dist.name();
+            let version = dist.version();
+            match minimum_version.entry(name) {
+                Entry::Vacant(entry) => {
+                    entry.insert(version);
+                }
+                Entry::Occupied(mut entry) => {
+                    if version < *entry.get() {
+                        entry.insert(version);
+                    }
+                }
+            }
+        }
+
+        // If any of the requirements were added without version specifiers, add a lower bound.
+        let mut modified = false;
+        for edit in &edits {
+            // Only set a minimum version for newly-added dependencies (as opposed to updates).
```

Comment:
Worth reviewing the criteria here for _when_ we set a lower bound: (1) never for existing dependencies, it must be a new dependency in `uv add`; (2) never when the user provides a URL.